lib/server/entries.js

```diff
@@ -99,8 +99,11 @@ function storage (env, ctx) {
 
       // Normalize dates to be in UTC, store offset in utcOffset
 
-      var _sysTime = moment(doc.dateString).isValid() ? moment.parseZone(doc.dateString) : moment(doc.date);
```

Comment:
The logic mistake here is - if dateString is missing or not valid, Moment defaults to the same behaviour as if just moment() was called, so the subsequent .isValid() calls are always truthy

Reply:
I looked at git blame; I think this code has been refactored so much that it's safe to remove any of this logic that does not make sense.  Git blame helped me recall that originally, about https://github.com/nightscout/cgm-remote-monitor/blame/f9d59c4e3e34c381b0925a60a5029f83d4e0d8e4/lib/server/entries.js#L83-L100, it was a major effort to switch from mills, which had bugs and issues in different timezones, to using any variant of 8601.  In addition, we were receiving two times system time and display time from the original G4 receivers, and hadn't decided which one to use yet.  In any case, over the years mobile uploaders have made syncing on universal time more trustworthy.